doc/reference/annotations.rst

```diff
@@ -463,6 +466,11 @@ Examples:
          */
         private $endAt;
 
+        /**
+         * @Type("DateTime<['Y-m-d', 'Y/m/d']>")

```

Comment:
This is not correct. Serialization does not accept more than one format. The feature you are building in only for de-serialization

Reply:
```
DateTime<'formats'>
```
is a shortcut for
```
DateTime<'formats', null, 'formats'>
```
that's why I allowed the syntax.

But for serialization I used the first valid format.

If I only allow the third parameter to be an array, it means that someone who only care bout de-serialization, will write
```
@Type("DateTime<'Y-m-d'>")
```
To deserialize only one format, and
```
@Type("DateTime<'Foo', 'Bar', ['Y-m-d', 'Y/m/d']>")
```
To handle multiple syntax. I found this a little sad...

Reply:
I can understand that, but `DateTime<['Y-m-d', 'Y/m/d']>` is wrong since the serialization does not accept multiple formats. it will always pick the first.

But to be hones, thinking it twice, I start to like this feature less and less. 

I had a similar problem in the past and i solved it with a custom date handler (see https://github.com/goetas-webservices/xsd2php-runtime/blob/master/src/Jms/Handler/XmlSchemaDateHandler.php#L86)

Reply:
I can only allow the syntax DateTime<'Foo', 'Bar', ['Y-m-d', 'Y/m/d']> if preferred.

> But to be hones, thinking it twice, I start to like this feature less and less.

It would be nice to make your choice, I took time working on it and I prefer to know now if I taking more time on it will be useless.

> I had a similar problem in the past and i solved it with a custom date handler

I feel like, it's really hard to write a custom date handler for someone starting with the library.
When only using the annotation (like we do in our company), it's so much easier to use an array that writing a whole custom dateHandler.